### PR TITLE
create first api for order service

### PR DESCRIPTION
**Pull request by LEHOANGGLAM**
create api: create order

### CODE REVIEW

File: order/src/main/resources/application.properties [hunk 1]
```diff
@@ -14,8 +14,8 @@ yas.services.customer=http://api.yas.local/customer
 
 spring.datasource.driver-class-name=org.postgresql.Driver
 spring.datasource.url=jdbc:postgresql://localhost:5432/order
-spring.datasource.username=admin
-spring.datasource.password=admin
+spring.datasource.username=postgres
+spring.datasource.password=123456
```
Comment (khoahd7621) [inline order/src/main/resources/application.properties:18]: Does it work? 
Database postgres exposed from docker has username is admin, and password is admin.

File: order/src/main/java/com/yas/order/service/OrderService.java [hunk 2]
```diff
@@ -1,18 +1,73 @@
...
-    public OrderService(OrderRepository orderRepository) {
+    public OrderService(OrderRepository orderRepository, OrderItemRepository orderItemRepository) {
         this.orderRepository = orderRepository;
+        this.orderItemRepository = orderItemRepository;
     }
 
+
+    public OrderVm createOrder(OrderPostVm orderPostVm) {
+        Validate.notNull(orderPostVm, "Order cannot be null");
+        Validate.notNull(orderPostVm.orderItemPostVms(), "Order Items cannot be null");
```
Comment (khoahd7621) [inline order/src/main/java/com/yas/order/service/OrderService.java:34]: You can use these annotations in the controller and record vm to validate these fields. Don't necessary to validate in the Services layer.

- Add annotation @Validated to Rest Controller class
- Add @Valid @NotNull before OrderPostVm params
- In record OrderPostVM you can also add @NotNull before OrderItemPostVms

**Review by khoahd7621 — COMMENTED** [review]
Reviewed, need some changes.

File: order/src/main/java/com/yas/order/repository/OrderItemRepository.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.yas.order.model.OrderItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+
+@Repository
+public interface OrderItemRepository extends JpaRepository<OrderItem, Long> {
+
+}
```
Comment (toannguyenk) [inline order/src/main/java/com/yas/order/repository/OrderItemRepository.java:11]: add blank line

File: order/src/main/java/com/yas/order/service/OrderService.java (hunk 2 above)
Comment (LEHOANGGLAM) [reply order/src/main/java/com/yas/order/service/OrderService.java:34]: thanks

File: order/src/main/java/com/yas/order/repository/OrderItemRepository.java (hunk 3 above)
Comment (LEHOANGGLAM) [reply order/src/main/java/com/yas/order/repository/OrderItemRepository.java:11]: thanks

**Review by chanhnguyen6975 — APPROVED** [review]

**Review by VoNhu2901 — APPROVED** [review]

**Pull request merged**